File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-8.java
class Solution {
    public int characterReplacement(String s, int k) {
        HashMap<Character, Integer> cnt = new HashMap<>();
        int res = 0;

        int l = 0, maxf = 0;
        for(int r = 0; r < s.length(); r++) {
            cnt.put(s.charAt(r), cnt.getOrDefault(s.charAt(r), 0) + 1);
            maxf = Math.max(maxf, cnt.get(s.charAt(r)));

            while((r - l + 1) - maxf > k) {
                cnt.put(s.charAt(l), cnt.get(s.charAt(l)) - 1);
                l++;
            }

            res = Math.max(res, r - l + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-8.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
       int res = -1, l = 0, r = heights.size() - 1;
       while(l < r) {
            int S = (r - l) * min(heights[l], heights[r]);
            res = max(res, S);
            if(heights[l] <= heights[r]) ++l;
            else --r;
       }
       return res;
    }
};
